src/main/java/com/amazonaws/encryptionsdk/internal/TrailingSignatureAlgorithm.java

```diff
@@ -66,37 +83,99 @@ public String getRawSignatureAlgorithm() {
             return hashAndSignAlgorithm;
         }
 
+        /**
+         * Decodes a compressed elliptic curve point as described in SEC-1 v2 section 2.3.4
+         * @see  <a href="http://www.secg.org/sec1-v2.pdf">http://www.secg.org/sec1-v2.pdf</a>
+         * @param keyString The serialized and compressed public key
+         * @return The PublicKey
+         */
         @Override
         public PublicKey deserializePublicKey(String keyString) {
-            final ECPoint q = ecSpec.getCurve().decodePoint(Utils.decodeBase64String(keyString));
-
-            ECPublicKeyParameters keyParams = new ECPublicKeyParameters(
-                    q,
-                    new ECDomainParameters(ecSpec.getCurve(), ecSpec.getG(), ecSpec.getN(), ecSpec.getH())
-            );
-
-            return new BCECPublicKey("EC", keyParams, ecSpec, BouncyCastleProvider.CONFIGURATION);
+            notNull(keyString, "keyString is required");
+
+            final byte[] decodedKey = Utils.decodeBase64String(keyString);
+            final BigInteger x = new BigInteger(1, Arrays.copyOfRange(decodedKey, 1, decodedKey.length));
+
+            final byte compressedY = decodedKey[0];
+            final BigInteger yOrder;
+
+            if(compressedY == TWO.byteValue()) {
+                yOrder = ZERO;
+            } else if (compressedY == THREE.byteValue()) {
+                yOrder = ONE;
+            } else {
+                throw new IllegalArgumentException("Compressed y value was invalid");
+            }
+
+            final BigInteger p = ((ECFieldFp)ecParameterSpec.getCurve().getField()).getP();
+            final BigInteger a = ecParameterSpec.getCurve().getA();
+            final BigInteger b = ecParameterSpec.getCurve().getB();
+
+            final BigInteger alpha = x.modPow(THREE, p)
```

Comment:
Please add a comment making it clear that alpha is intended to be y^2 (and that we validate this later)

Reply:
Added